### PR TITLE
Fix testFilterWithUdf

### PR DESCRIPTION
**Pull request by elonazoulay**
This is due to differences in how Pinot handles IEEE-754 approximate numerics pre/post Pinot 0.12.1.

See https://github.com/apache/pinot/issues/10637

## Release notes

(X ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestDynamicTable.java [hunk 1]
```diff
@@ -183,7 +183,9 @@ public void testFilterWithPushdownConstraint()
     public void testFilterWithUdf()
```
Comment (kokosing) [inline plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestDynamicTable.java:183]: Can you please fix pinot tests to use capital letters for SQL keywords like: SELECT, FROM, WHERE? Let's do it as separate Pr.

Comment (kokosing) [inline plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestDynamicTable.java:183]: Can you please let me know what are you fixing here? Is this a flaky issue? Do you have any github issues reported for the problem you are fixing? How the problem you are fixing manifests?

Comment (elonazoulay) [reply plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestDynamicTable.java:183]: Sure! Will do.

Comment (elonazoulay) [reply plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestDynamicTable.java:183]: Created #17109

**Comment by hashhar** [comment]
before merging this we need to verify if the Pinot results actually depend on JVM version being used instead of the Pinot version itself.

**Comment by elonazoulay** [comment]
> before merging this we need to verify if the Pinot results actually depend on JVM version being used instead of the Pinot version itself.

I tried this on java11 and java17 and they behave the same. Changing Pinot version results in the different behavior. Also updated the pinot issue with those findings.

**Comment by elonazoulay** [comment]
Also, adding `.1` to the query in the test guarantees that the actual and expected result will be equal regardless of the version.

lmk if you need more info.

**Review by hashhar — APPROVED** [review]
Indeed, I can see the different behaviour as well. Thanks for including repro steps.

LGTM.
